src/main/java/org/jboss/staxmapper/XMLMapperImpl.java

```diff
@@ -100,10 +108,14 @@ public void deparseDocument(XMLContentWriter contentWriter, XMLStreamWriter stre
         contentWriter.writeContent(new FormattingXMLStreamWriter(streamWriter));
     }
 
-    @SuppressWarnings({ "unchecked" })
+    private XMLElementReader<?> getParser(final QName name) {
+        return rootElements.computeIfAbsent(name, qName -> rootElementsProvider.get(name).get());
```

Comment:
If the map's get() fails, will it not result in an NPE?  Maybe this should be ``rootElementsProvider.getOrDefault(name, () -> null).get()`` instead.